client/src/app/api/models.tsx

```diff
@@ -400,7 +405,7 @@ export interface Taskgroup {
   id?: number;
   name: string;
   addon: string;
-  data: TaskData;
+  data: any; // TODO(djzager) - may want to fix this
```

Comment:
Any reason not to use the TaskData type here?

Reply:
This point still remains but the other issues have been addressed.